### PR TITLE
fix: [PL-42846]: Add description field to APIKeyCard

### PR DESCRIPTION
**Pull request by ishantmehtaharness**
-- Added Description field to the API key list
-- tested at multiple places
-- No impact

Screenshot 2023-11-20 at 10 24 36 PM

Screenshot 2023-11-20 at 10 24 50 PM

### CODE REVIEW

File: src/modules/20-rbac/pages/ServiceAccounts/views/ServiceAccountsListView.tsx [hunk 1]
```diff
@@ -66,6 +68,19 @@ const RenderColumnDetails: Renderer<CellProps<ServiceAccountAggregateDTO>> = ({
           <Text color={Color.BLACK} lineClamp={1} className={css.wordBreak}>
             {data.name}
           </Text>
+          {data.description ? (
```
Comment (sahithikolichala) [inline src/modules/20-rbac/pages/ServiceAccounts/views/ServiceAccountsListView.tsx:71]: This code seems repetitive, may be make this as a component and use it at both places

File: src/modules/20-rbac/components/ApiKeyList/views/ApiKeyCard.tsx [hunk 2]
```diff
@@ -47,6 +48,19 @@ const RenderColumnDetails = (data: ApiKeyDTO): React.ReactElement => {
         <Text font={{ variation: FontVariation.BODY2 }} color={Color.BLACK} lineClamp={1} className={css.wordBreak}>
           {data.name}
         </Text>
+        {data.description ? (
```
[the remark this answers is not in the document]
Comment (ishantmehtaharness) [reply src/modules/20-rbac/components/ApiKeyList/views/ApiKeyCard.tsx:51]: Not needed

File: src/modules/20-rbac/pages/ServiceAccounts/views/ServiceAccountsListView.tsx (hunk 1 above)
Comment (ishantmehtaharness) [reply src/modules/20-rbac/pages/ServiceAccounts/views/ServiceAccountsListView.tsx:71]: Yes, that sounds good. updated

**Review by pankajsingh-harness — DISMISSED** [review]

**Comment by ishantmehtaharness** [comment]
trigger eslint

**Comment by ishantmehtaharness** [comment]
trigger eslint

**Comment by ishantmehtaharness** [comment]
retrigger eslint

File: src/modules/20-rbac/components/DescriptionPopover/__tests__/DescriptionPopover.test.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  })
+}))
+describe('DescriptionPopover', () => {
+  test('empty description props', () => {
+    const { container } = render(<DescriptionPopover description={''} />)
+    expect(container.querySelector('[data-icon="description"]')).toBeNull()
+  })
+  test('valid description props', () => {
+    const { container } = render(<DescriptionPopover description={'Hey this is description'} />)
+    expect(container.querySelector('[data-icon="description"]')).not.toBeNull()
```
Comment (pankajsingh-harness) [inline src/modules/20-rbac/components/DescriptionPopover/__tests__/DescriptionPopover.test.tsx:24]: NIT: Please check for correct data

**Review by pankajsingh-harness — APPROVED** [review]

**Review by namanharness — APPROVED** [review]

File: src/modules/20-rbac/components/DescriptionPopover/__tests__/DescriptionPopover.test.tsx (hunk 3 above)
Comment (ishantmehtaharness) [reply src/modules/20-rbac/components/DescriptionPopover/__tests__/DescriptionPopover.test.tsx:24]: data comes on hover
we can add one more test to check that too

**Pull request merged**